klog.go

```diff
@@ -824,21 +831,50 @@ func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
 		}
 		b.WriteByte(' ')
 
-		switch v.(type) {
-		case string, error:
-			b.WriteString(fmt.Sprintf("%s=%q", k, v))
+		switch v := v.(type) {
+		case string:
+			writeStringValue(b, k, true, v)
+		case error:
+			writeStringValue(b, k, true, v.Error())
 		case []byte:
 			b.WriteString(fmt.Sprintf("%s=%+q", k, v))
 		default:
-			if _, ok := v.(fmt.Stringer); ok {
-				b.WriteString(fmt.Sprintf("%s=%q", k, v))
+			if s, ok := v.(fmt.Stringer); ok {
+				writeStringValue(b, k, true, s.String())
 			} else {
-				b.WriteString(fmt.Sprintf("%s=%+v", k, v))
+				writeStringValue(b, k, false, fmt.Sprintf("%+v", v))
 			}
 		}
 	}
 }
 
+func writeStringValue(b *bytes.Buffer, k interface{}, quoteV bool, v string) {
+	if !strings.Contains(v, "\n") {
+		b.WriteString(fmt.Sprintf("%s=", k))
+		if quoteV {
+			// Simple string, quote quotation marks and non-printable characters.
+			b.WriteString(strconv.Quote(v))
+			return
+		}
+		// Non-string with no line breaks.
+		b.WriteString(v)
+		return
+	}
+	// Complex multi-line string, show as-is with indention.
```

Comment:
I don't see any benefit of adding built in markers for logs. They 
* don't make logs more readable as users can klog header to already distinguish start vs end of multi line log
* adding standard markers makes parsing logs harder, existing kubernetes log parsers already should handle multi line log.
* Increases log volume

K8s logs already have multi line logs that provide their own markers, thus I think we should leave it up user to provide 
them themselves.

Reply:
I think there's a misunderstanding. The markers are not delimiters for *messages*. They are delimiters for *values*.

With the current InfoS format, the quotation marks provides those delimiters. But they result in output that is basically unreadable for a human, which defeats the purpose of the text output.

Reply:
> Increases log volume

That's a valid point. We could make them shorter. For example:
```
I1112 15:01:48.696677  413871 structured_logging.go:52] using InfoS with
 the message across multiple lines: int=1 stringData>>>
 long: Multiple
 lines
 with quite a bit
 of text.
 <<< str="another value"
```
That's four bytes (`>>>` plus newline) instead of two (`="`) and three bytes (`<<<`) instead of one (`"`) at the end.

Line breaks are the same with new line plus space instead of `\\n`.

If the text contains many quotation marks, this format might actually be shorter because those don't need to be escaped.